### ISSUE TITLE
Promise created but not returned: False-Positive?

### ISSUE THREAD

**Issue by nmagerko** [petkaantonov/bluebird#1141]
1) Bluebird version: v3.4.1 (latest)
2) Platform version:  Node.js v4.4.3 + Express v4.14.0
3) Did this issue happen with earlier version of bluebird?
Unknown, although I have seen a few issues related to this.

I have a stack trace that looks like this (truncated for brevity):
```
Warning: a promise was created in a handler but was not returned from it
    at processImmediate [as _immediateCallback] (timers.js:383:17)
From previous event:
    at Runner.ensureConnection (.../node_modules/knex/lib/runner.js:186:39)
    at Runner.run (.../node_modules/knex/lib/runner.js:41:44)
    at QueryBuilder.Target.then (.../node_modules/knex/lib/interface.js:32:43)
    at Object.module.exports.findById (.../services/Service.js:57:4)
...
From previous event:
    at module.exports (.../middleware/auth.js:14:4)
...
```

The first event is referencing part of a service of mine that makes use of the `knex` NPM module. However, as there are no issues on that repository's issue tracker related to this (that I can find), I'm assuming it has something to do with the code I've written elsewhere.

So, looking at the event before that, I find a call made in my `auth.js` sub-module. Line 14 is a `then` which expects a result from the following auth verification service method:
```
var Promise = require('bluebird');
var jwt = require('jsonwebtoken');

module.exports.verify = function(token) {
	return Promise
		.resolve(jwt.verify(token, SECRET))
		.catch(jwt.JsonWebTokenError, function (error) {
			var message = error.message;
			throw new SpecializedError(message);
		});
};
```

In other words, I am returning a resolved promise result, where the result is produced synchronously. I then add a catch-handler onto the promise before returning, rethrowing the error as a special type. If it helps, the calling `then` itself looks like this:

```
Service.verify(auth)
	.then(function (decoded) {
		req.auth = decoded; // express request object
		next(); // express next method
	})
	.catch(function (error) {
		// handle error
	});
``` 

It seems like this function is causing the warning, although I'm almost positive that this is a valid return of a promise. The `then` which handles it seems to be working properly, and I can't seem to find anything else that would cause the warning to be propagated. 

Thoughts?

**Comment by phpnode** [comment]
Add a `return null;` at the end of your `.then()` call.

**Comment by nmagerko** [comment]
That was the issue. I guess I didn't realize that's the portion of the call it was referring to. Thanks!
